File: product-of-array-except-self/product-of-array-except-self.cpp
class Solution {
public:
    vector<int> productExceptSelf(vector<int>& nums) {
        int n = nums.size() ;
        vector<int>res;
        int ans = 1 ;
        for(int i = 0 ; i < n ; i++){
            ans *= nums[i];
            res.push_back(ans);
        }
        ans = 1 ;

        for(int i = n-1 ; i>0 ; i--){
            res[i] = res[i-1]*ans;
            ans *= nums[i];
        }
        res[0] = ans ;

        return res ;
    }
};

File: product-of-array-except-self/product-of-array-except-self.py
class Solution:
    def productExceptSelf(self, nums: List[int]) -> List[int]:
        res = []
        ans = 1 
        n = len(nums)
        for i in range(n):
            ans *= nums[i]
            res.append(ans)
        ans =1 
       
        for i in range(n-1,0,-1):
            res[i] = res[i-1]*ans
            ans *= nums[i]
        res[0] = ans
        return res
